### DISCUSSION TITLE
Switch/button configuration that acts like momentary button, unless held down/open?

### DISCUSSION THREAD

**Discussion by Jibun-no-Kage** [arendst/Tasmota#17655] [Q&A]
Switch/button configuration that acts like momentary button, unless held down, held down/open for a set time?  I have an ESP8266 that is driving a mosfet based controller.  I need a switch or button configuration that acts like a momentary button, unless explicitly held down/open?

Can I have two controls in the tasmota UI, that drive the same GPIO pin?  That seems possible from what I have read.  Am I correct on this?  Such that if I define in the template two methods to control the same GPIO?

The first would be a simple momentary button setup.  The second would hold the button down/open for  a set time, say a few seconds, when activated?

I know I could drive this logic from an application or something like Node Red, but wonder if the logic can actually be implemented in the actual direct Tasmota web based interface/UI?

**Comment by barbudor** [comment]
Sorry I don't understand your use case
Could you please describe in plain English words what you would like to happen?

**Comment by Jibun-no-Kage** [comment]
Using just the Tasmota firmware and using only associated on-board UI:

Have one UI control drive a given GPIO pin (that happens to be connect to a mosfet based control board) between high and return to low state.  Then have another UI control drive the same GPIO pin to high for a time (say 5 seconds) then back to low.

So in the first use case, it just a momentary button behavior.  In the additional use case, pushing/clicking the UI control sets high state, then a delay occurs, then sets low state.

I think the first use case is straight forward for Tasmota configuration/UI, but the second use case is not.

**Reply by barbudor** [reply]
By "UI", are you refering to Web UI ?
Web UI doesn't support momentary push button. A Web button always send a POWER TOGGLE command

For hardware buttons/switch
You can have a Switch_1 set in "Follow mode" which is `Switchmode1 1` (or `Switchmode1 2` depending of the cabling of your switch - if one does the opposite of what you want, try the other)
With this configuration, the output Relay1 will follow the switch : when you press the switch, Relay1 output will be HIGH, when you release the switch Relay1 output will be LOW

Then you can have another Switch_2 in the same mode `Switchmode2 3` and use a rule with a RuleTimer such as:
```
Rule1 on Switch2#state do backlog Power1 ON; RuleTimer1 5 endon on Rules#Timer=1 do Power1 OFF
Rule1 1
```

Hopefully I have properly understood your need

**Comment by Jibun-no-Kage** [comment]
Nat all you pretty much have it right.  Of course one can do this external to the native web UI via say Node Red, which I have done.  But I wanted to at least try to do something that was 100% within Tasmota.  I will give what you suggest a try to see how it works.  I guess I am a bit surprised a momentary push button feature is not implemented?  But given the core design is the drive relays, and such I can how that might be a less use case.  But that said, would it not be a nice feature to have?
